circuit_knitting/cutting/cutting_evaluation.py

```diff
@@ -393,9 +473,13 @@ def _get_bases_by_partition(
     bases_dict = {}
     for i, subcirc in enumerate(subcirc_qpd_gate_ids):
         for basis_id in subcirc:
-            decomp_id = int(
-                circuits[i].data[basis_id[0]].operation.label.split("_")[-1]
-            )
+            try:
+                decomp_id = int(
+                    circuits[i].data[basis_id[0]].operation.label.split("_")[-1]
+                )
+            except (AttributeError, ValueError):  # pragma: no cover
+                _raise_bad_qpd_gate_labels()  # pragma: no cover
```

Comment:
Is there really no way to test these lines?

Reply:
I think I was seeing bad behavior from coverage here, let me get back to you

Reply:
Fixed this by removing the check in one private function, which is called strictly after another private function which makes the same check. The other option was to check that private function itself in a test, which seemed overkill.

Also added one test to test the case in the first private function that should catch the bad labels.